pkg/datarepair/datarepair_test.go

```diff
@@ -125,7 +148,147 @@ func TestDataRepair(t *testing.T) {
 
 		remotePieces = pointer.GetRemote().GetRemotePieces()
 		for _, piece := range remotePieces {
-			assert.False(t, nodesToKill[piece.NodeId])
+			require.False(t, nodesToKill[piece.NodeId])
+			require.False(t, nodesToDisqualify[piece.NodeId])
+		}
+	})
+}
+
+// TestRepairMultipleDisqualified does the following:
+// - Uploads test data to 7 nodes
+// - Disqualifies 3 nodes
+// - Triggers data repair, which repairs the data from the remaining 4 nodes to additional 3 new nodes
+// - Shuts down the 4 nodes from which the data was repaired
+// - Now we have just the 3 new nodes to which the data was repaired
+// - Downloads the data from these 3 nodes (succeeds because 3 nodes are enough for download)
+func TestRepairMultipleDisqualified(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount:   1,
+		StorageNodeCount: 12,
+		UplinkCount:      1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		// first, upload some remote data
+		ul := planet.Uplinks[0]
+		satellite := planet.Satellites[0]
+		// stop discovery service so that we do not get a race condition when we delete nodes from overlay cache
+		satellite.Discovery.Service.Discovery.Stop()
+		satellite.Discovery.Service.Refresh.Stop()
+
+		satellite.Repair.Checker.Loop.Pause()
+		satellite.Repair.Repairer.Loop.Pause()
+
+		testData := make([]byte, 1*memory.MiB)
+		_, err := rand.Read(testData)
+		require.NoError(t, err)
```

Comment:
There's testrand package that makes this part more convenient. Might have gotten accidentally reverted.